File: sync_anich.js
const fs = require('fs');
const path = require('path');

// 将数据直接备份到 web 根目录下，由 Git 跟踪进行物理发布
const dataDir = './web';

async function fetchAndSave(url, fileName) {
  try {
    console.log(`[Node.js 抓取] 正在请求 ${url} ...`);
    const res = await fetch(url, {
      headers: {
        'User-Agent': 'xs IOS 1.0.0',
        'Accept': 'application/json'
      }
    });
    if (res.status === 200) {
      const data = await res.json();
      if (!fs.existsSync(dataDir)) {
        fs.mkdirSync(dataDir, { recursive: true });
      }
      fs.writeFileSync(path.join(dataDir, fileName), JSON.stringify(data, null, 2));
      console.log(`[成功] 备份 ${fileName}`);
      return true;
    } else {
      console.error(`[Error] 备份 ${fileName} 失败: HTTP Status ${res.status}`);
    }
  } catch (e) {
    console.error(`[Error] 备份 ${fileName} 发生异常:`, e.message);
  }
  return false;
}

async function main() {
  console.log("============== [开始本地 Node.js 数据备份] ==============");
  
  // 1. 抓取最新更新 (Latest)
  await fetchAndSave(
    'https://api.emmmm.eu.org/latest?sort=-1&type=all&skip=0',
    'latest.json'
  );

  // 2. 抓取全局番剧大表
  await fetchAndSave(
    'https://api.emmmm.eu.org/bangumi/list?skip=0&type=all',
    'bangumi_list.json'
  );

  console.log("============== [本地 Node.js 数据备份完成] ==============");
}

main();

File: sync_anich.py
import subprocess
import json
import time
import os
import urllib.parse

# 直接创建在最终编译输出的 build/web 目录下，彻底避开 Flutter 编译器的文件过滤
data_dir = "./build/web"
os.makedirs(data_dir, exist_ok=True)

def fetch_json(url):
    try:
        # 将原始 URL 进行 UrlEncode 编码，套上 AllOrigins 进行云端中转抓取，彻底规避 TLSV1_ALERT_INTERNAL_ERROR
        proxy_url = f"https://api.allorigins.win/raw?url={urllib.parse.quote_plus(url)}"
        result = subprocess.run(
            ['curl', '-k', '-L', '-s', '-A', 'xs IOS 1.0.0', proxy_url],
            capture_output=True, text=True, timeout=20
        )
        if result.returncode == 0 and result.stdout.strip():
            return json.loads(result.stdout)
    except Exception as e:
        print(f"[Error] Failed to fetch {url} via proxy curl: {e}")
    return None

print("============== [开始云端数据备份] ==============")

# 1. 抓取最新更新 (Latest)
latest_url = "https://api.emmmm.eu.org/latest?sort=-1&type=all&skip=0"
latest_data = fetch_json(latest_url)
if latest_data:
    with open(f"{data_dir}/latest.json", 'w', encoding='utf-8') as f:
        json.dump(latest_data, f, ensure_ascii=False, indent=2)
    print("[成功] 备份 latest.json")

# 2. 抓取全局番剧大表，用于轻应用前端的免跨域“极速检索”
list_url = "https://api.emmmm.eu.org/bangumi/list?skip=0&type=all"
list_data = fetch_json(list_url)
if list_data:
    with open(f"{data_dir}/bangumi_list.json", 'w', encoding='utf-8') as f:
        json.dump(list_data, f, ensure_ascii=False, indent=2)
    print("[成功] 备份 bangumi_list.json (用于本地全局搜索)")

print("============== [云端数据备份完成] ==============")
